### PR TITLE
Add solutions for Module 1 home tasks

### PR DESCRIPTION
**Pull request by apankratau**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,12 @@
   "description": "NodeJS Global Mentoring Program Hometasks",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "task-1.1": "nodemon module-1/task-1",
+    "task-1.2": "nodemon module-1/task-2",
+    "build-task-1.1": "babel module-1/task-3/task-1 -d module-1/task-3/task-1/dist",
```
Comment (Dmitry-White) [inline package.json]: Typically, when just a Vanilla JS is used, there's no need to create a build on the server.
The server process is launched and it handles the transformations (such as babel) on the fly.
Therefore, you could've used babel-node and instruct nodemon to use it on the fly using "--exec".
https://www.codementor.io/@michaelumanah/how-to-set-up-babel-7-and-nodemon-with-node-js-pbj7cietc

File: module-1/task-1/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const readline = require('readline');
+
+const processLineByLine = () => {
+  const rl = readline.createInterface({
```
Comment (Dmitry-White) [inline module-1/task-1/index.js]: Creating an instance of readline could be done outside of a running handler function so that it could be reused in more than one handler.
In your case, calling processLineByLine is redundant since Node.js executes the whole file anyway.
In a typical JS program the purpose of creating function is to reuse parts of the logic or call the function recursively.
In your case, calling processLineByLine will recreate the readline interface and reassign event listeners which could potentially lead to memory leaks.

File: module-1/task-2/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+      return newObj;
+    }, {});
+};
+
+const main = () => {
+  pipeline(
+    fs.createReadStream(PATH_TO_CSV),
+    csv(),
+    new Transform({
```
Comment (Dmitry-White) [inline module-1/task-2/index.js]: By using Transform here you're essentially storing parts of the stream as Buffers (aka in RAM memory), violating the task requirements.
Moreover, it's a bit unclear why would you need to use transform to accomplish the task that is already handled by csv() method.
The case sensitiveness issue is cause by using quotes around every value in csv file, which is note the case in a typical csv file (the content is already a string)

File: module-1/task-2/index.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      if (err) {
+        console.error('Pipeline failed', err);
+      } else {
+        console.log('Pipeline succeeded');
+      }
+    }
+  );
+};
+
+main();
```
Comment (Dmitry-White) [inline module-1/task-2/index.js]: And again, there's really no need to wrap the code in a function call :)

**Review by Dmitry-White — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (apankratau) [reply package.json]: Fixed, thanks!

File: module-1/task-1/index.js (hunk 2 above)
Comment (apankratau) [reply module-1/task-1/index.js]: Fixed, thanks!

File: module-1/task-2/index.js (hunk 4 above)
Comment (apankratau) [reply module-1/task-2/index.js]: Removed, thanks

File: module-1/task-2/index.js (hunk 3 above)
Comment (apankratau) [reply module-1/task-2/index.js]: So I guess I overcomplicated this thing a little bit. Removed the transformation at all, seems to work as expected. However what if I really need to transform the resulting object somehow? How should that be done not to bloat RAM memory etc? As far as I remember at first I used [.subscribe](https://github.com/Keyang/node-csvtojson#result-transform) method from csv library itself but it didn't work, the code inside this method just never got executed, I assume because calling `csv()` just returned a stream downwards. Then I switched to using `Transform` and that seemed to solve the issue

File: package.json (hunk 1 above)
Comment (apankratau) [reply package.json]: Fixed, thanks!

File: module-1/task-1/index.js (hunk 2 above)
Comment (apankratau) [reply module-1/task-1/index.js]: Fixed, thanks!

File: module-1/task-2/index.js (hunk 3 above)
Comment (apankratau) [reply module-1/task-2/index.js]: So I guess I overcomplicated this thing a little bit. Removed the transformation at all, seems to work as expected. However what if I really need to transform the resulting object somehow? How should that be done not to bloat RAM memory etc? As far as I remember at first I used [.subscribe](https://github.com/Keyang/node-csvtojson#result-transform) method from csv library itself but it didn't work, the code inside this method just never got executed, I assume because calling `csv()` just returned a stream downwards. Then I switched to using `Transform` and that seemed to solve the issue

File: module-1/task-2/index.js (hunk 4 above)
Comment (apankratau) [reply module-1/task-2/index.js]: Removed, thanks

**Review by Dmitry-White — APPROVED** [review]
Nice work!

File: module-1/task-2/index.js (hunk 3 above)
Comment (Dmitry-White) [reply module-1/task-2/index.js]: In general, you're correct in your assumption that if a transform is needed you need to store at least something in the memory to perform the necessary operations. 
While streaming data you're basically breaking it into "chunks". Depending on the type of the transformation and the data itself you either perform the operations on the whole data (once its chunks are all in memory) or on the individual chunks.
`data = a + b` => `T*data = T1*a + T2*b` OR `T*data = T0*(a + b)`;
To put it simply: 
If the data is homogeneous (an image) - you can transform them chunk by chunk and the memory consumption stays low (like image resizing). 
If the data is heterogeneous - you'll need to collect the whole thing and apply processing.

Comment (Dmitry-White) [reply module-1/task-2/index.js]: But the task was simple - just stream data from one place to another (one format to another) not really caring about its content :)
